test/hotspot/jtreg/compiler/vectorization/TestFloatConversionsVector.java

```diff
@@ -26,7 +26,9 @@
  * @bug 8294588
  * @summary Auto-vectorize Float.floatToFloat16, Float.float16ToFloat APIs
  * @requires vm.compiler2.enabled
- * @requires (os.simpleArch == "x64" & (vm.cpu.features ~= ".*avx512f.*" | vm.cpu.features ~= ".*f16c.*")) | os.arch == "aarch64"
+ * @requires (os.simpleArch == "x64" & (vm.cpu.features ~= ".*avx512f.*" | vm.cpu.features ~= ".*f16c.*")) |
+ *           os.arch == "aarch64" |
+ *           (os.arch == "riscv64" & vm.cpu.features ~= ".*zvfh.*")
```

Comment:
A small question: I see you used following condition when doing the scalar version:
    `(os.arch == "riscv64" & vm.cpu.features ~= ".*zfh,.*")`
There is comma there for matching cpu features. Do we really need this comma? It is not there for this vector version.

Reply:
Seems not necessary currently, unless in the future there is other extension starting with `zfh`, e.g. `zfhX`, but seems that chance is low IMHO.
And, I think it might not work for the last extension in the CPU feature string if it's with a comma (as the cpu feature string is converted from a list to string, so the last one will not ending with a`,` ? ), so for new and long named extensions, it should be without a comma.

Reply:
Thanks for testing!

Reply:
> Seems not necessary currently, unless in the future there is other extension starting with `zfh`, e.g. `zfhX`, but seems that chance is low IMHO. And, I think it might not work for the last extension in the CPU feature string if it's with a comma (as the cpu feature string is converted from a list to string, so the last one will not ending with a`,` ? ), so for new and long named extensions, it should be without a comma.

Yeah, that makes sense to me. I think we should remove this comma from conditions `(os.arch == "riscv64" & vm.cpu.features ~= ".*zfh,.*")` on jdk master. Can you fix that? Then we will have a more simpler rule when writing those conditions. That is, for single-character extensions like `c` or `v`, we need to add this comma. But for multi-character extensions like `zfh` or `zvfh`, we should not have it to avoid the issue you mentioned.

Reply:
Sure, I created https://bugs.openjdk.org/browse/JDK-8327689 to track it.